aries_cloudagent/protocols/routing/v1_0/handlers/forward_handler.py

```diff
@@ -38,8 +38,20 @@ async def handle(self, context: RequestContext, responder: BaseResponder):
             self._logger.exception("Error resolving recipient for forwarded message")
             return
 
+        # load connection
+        connection_mgr = ConnectionManager(context)
+        connection_targets = await connection_mgr.get_connection_targets(
+            connection_id=recipient.connection_id
+        )
+        # TODO: validate that there is 1 target, with 1 verkey. warn otherwise
+        connection_verkey = connection_targets[0].recipient_keys[0]
+
         # Note: not currently vetting the state of the connection here
         self._logger.info(
             f"Forwarding message to connection: {recipient.connection_id}"
         )
-        await responder.send(packed, connection_id=recipient.connection_id)
+        await responder.send(
+            packed,
+            connection_id=recipient.connection_id,
```

Comment:
Looking at the responder code, I think you can pass `connection_targets` as `target_list`, instead of passing `connection_id`, and it would avoid the need to look up the connection information twice.